src/ZLogger.Unity/Assets/Scripts/ZLogger/AsyncStreamLineMessageWriter.cs

```diff
@@ -1,62 +1,212 @@
+        readonly ZLoggerOptions          options;
         readonly CancellationTokenSource cancellationTokenSource;
 
         public AsyncStreamLineMessageWriter(Stream stream, ZLoggerOptions options)
         {
-            this.newLine = Encoding.UTF8.GetBytes(Environment.NewLine);
+            this.newLine                 = Encoding.UTF8.GetBytes(Environment.NewLine);
             this.cancellationTokenSource = new CancellationTokenSource();
             if (newLine.Length == 1)
             {
                 // cr or lf
                 this.newLine1 = newLine[0];
                 this.newLine2 = default;
-                this.crlf = false;
+                this.crlf     = false;
             }
             else
             {
                 // crlf(windows)
                 this.newLine1 = newLine[0];
                 this.newLine2 = newLine[1];
-                this.crlf = true;
+                this.crlf     = true;
             }
 
             this.options = options;
-            this.stream = stream;
+            this.stream  = stream;
             this.channel = Channel.CreateUnbounded<IZLoggerEntry>(new UnboundedChannelOptions
             {
                 AllowSynchronousContinuations = false, // always should be in async loop.
-                SingleWriter = false,
-                SingleReader = true,
+                SingleWriter                  = false,
+                SingleReader                  = true,
             });
-            this.writeLoop = Task.Run(WriteLoop);
+
+            this.writeLoop        = Task.Run(WriteLoop);
+            this.summaryWriteLoop = Task.Run(SummaryWriteLoop);
         }
 
+        private const bool ENABLE_SPAM_DROPPER  = true;
+        private const bool DEBUG_SPAM           = false;
+        private const uint LIMIT_IN             = 3;
+        private const uint LIMIT_OUT            = 2;
+        private const uint POSTS_SECONDS_WINDOW = 1;
+        private       bool isSpamming;
+        private       bool didDrop;
+
+        private readonly ConcurrentDictionary<LogLevel, int> dropSummary = new(new KeyValuePair<LogLevel, int>[]
+        {
+            new(LogLevel.Trace, 0),
+            new(LogLevel.Debug, 0),
+            new(LogLevel.Information, 0),
+            new(LogLevel.Warning, 0),
+            new(LogLevel.Error, 0),
+            new(LogLevel.Critical, 0),
+            new(LogLevel.None, 0),
+        });
+
+        private readonly ConcurrentQueue<DateTimeOffset> postTimesQ                  = new();
+        private readonly ConcurrentQueue<DateTimeOffset> postTimesTempQ              = new();
+        private readonly ArrayBufferWriter<byte>         bufferWriterForDebugSpamMsg = new();
+        
+        private static readonly object lockObject = new();
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Post(IZLoggerEntry log)
         {
-            channel.Writer.TryWrite(log);
+            // lock (lockObject)
```

Comment:
no thanks

Reply:
79b11fa - remove lock comment and braces